src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,84 @@
+package duke;
+
+import java.util.Scanner;
+
+public class Duke {
+    private TaskList tasks;
+    private Storage storage;
+    private Ui ui;
+    private Parser parser;
+
+    public Duke() {
+        this.tasks = new TaskList();
+        this.storage = new Storage("C:/Users/Jeremias/Documents/GitHub/ip/data/", "data.txt");
+        this.ui = new Ui();
+        this.parser = new Parser();
+    }
+
+    public static void main(String[] args) {
+        Duke duke = new Duke();
+        duke.run();
+    }
+
+    /**
+     * Starts the Duke program.
+     */
+    private void run() {
+        ui.printGreeting();
+        storage.loadTaskList(tasks);
+        Scanner sc = new Scanner(System.in);
+        boolean exit = false;
+        while (!exit) {
+            String input = sc.nextLine();
+            if (input.equals("bye")) {
+                exit = true;
+                ui.printBye();
+            } else {
+                executeCommand(input);
+                save();
+            }
+        }
+    }
+
+    /**
+     * Executes the command that the user inputs.
+     *
+     * @param input User input.
+     */
+    private void executeCommand(String input) {
+        String command = parser.parseCommand(input);
+        Command cmd = null;
+        try {
+            if (command.equals("todo") || command.equals("deadline") || command.equals("event")) {
```

Comment:
I think a cleaner alternative to having such a long if expression here would be to use a switch statement instead. E.g.

```java
switch (command) {
case "todo":
    // Fallthrough
case "deadline":
    // Fallthrough
case "event":
    // Fallthrough
    cmd = new AddTaskCommand(command, input, tasks);
    break;
case "list":
...
```